synapseclient/client.py

```diff
@@ -626,6 +626,27 @@ def is_certified(self, user: typing.Union[str, int]) -> bool:
                 return False
             raise
 
+    def is_synapse_id(self, syn_id):
+        """Checks if given synID is valid (attached to actual entity?)"""
+        if isinstance(syn_id, str):
+            try:
+                try:
+                    self.get(syn_id, downloadFile=False)
+                except SynapseFileNotFoundError:
+                    return False
+                except SynapseHTTPError as err:
+                    if err.response.status_code in (400, 404):
+                        return False
+                    # Valid ID but user lacks permission
+                    elif err.response.status_code == 403:
+                        return True
+            except SynapseAuthenticationError as err:
```

Comment:
@vpchung what is the reason that there isn't just three `except` calls?

```
try:
   ...
except SynapseFileNotFoundError:
   ...
except SynapseHTTPError as err:
   ...
except SynapseAuthenticationError as err:
   ....
```

Reply:
Ah, this is how I usually catch exception chains.  But yes, you are right -- for this particular scenario, we don't need to be this differential with the catches.